Services/Database/classes/Setup/class.ilDatabaseUpdatedObjective.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/* Copyright (c) 2019 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+use ILIAS\Setup;
+use ILIAS\DI;
+
+class ilDatabaseUpdatedObjective extends \ilDatabaseObjective {
+	public function getHash() : string {
+		return hash("sha256", implode("-", [
+			self::class,
+			$this->config->getHost(),
+			$this->config->getPort(),
+			$this->config->getDatabase()
+		]));
+	}
+
+	public function getLabel() : string {
+		return "The database is updated.";
+	}
+
+	public function isNotable() : bool {
+		return true;
+	}
+
+	public function getPreconditions(Setup\Environment $environment) : array {
+		$common_config = $environment->getConfigFor("common");
+		return [
+			new \ilIniFilesPopulatedObjective($common_config),
+			new \ilDatabasePopulatedObjective($this->config)
+		];
+	}
+
+	public function achieve(Setup\Environment $environment) : Setup\Environment {
+		$db = $environment->getResource(Setup\Environment::RESOURCE_DATABASE);
+		$io = $environment->getResource(Setup\Environment::RESOURCE_ADMIN_INTERACTION);
+		$client_ini = $environment->getResource(Setup\Environment::RESOURCE_CLIENT_INI);
+		$common_config = $environment->getConfigFor("common");
+		$filesystem_config = $environment->getConfigFor("filesystem");
+
+		// ATTENTION: This is a total abomination. It only exists to allow the db-
```

Comment:
Thanks for clarifying this, since we will have to tackle this later. did you already test a full upgrade vom e.g.5.3 to 6?

Reply:
No, currently I only updated from 5.4 to 6. I put this on my agenda. Also, I'm thinking about invoking some admins to test this thoroughly. Maybe @corro want to have look?

Reply:
Good idea, I'll have a look at it with @corro tomorrow!